### DISCUSSION TITLE
在使用了TLS的情况下还有必要使用加密吗？

### DISCUSSION THREAD

**Discussion by gitcoinkuang** [v2fly/v2ray-core#2179] [General]
协议使用vmess+ws+tls，在使用的时候加密选择none或者zero应该没问题把？tls已经加密了，应该无法解密，假定证书永远需要验证

**Comment by qiangliu8183** [comment]
没问题，文档好像推荐zero

**Reply by gitcoinkuang** [reply]
文档是推荐auto，说是兼容性和安全性最高，zero不加密也不验证，但速度理论最快
